File: buck_tanley_app/lib/models/dto/LoginDTO.dart
class LoginDTO {
  final String userId;
  final String userPw;
  final String platform;
  final String? fcmToken;

  LoginDTO({
    required this.userId,
    required this.userPw,
    required this.platform,
    required this.fcmToken,
  });

  // JSON -> 객체 변환
  factory LoginDTO.fromJson(Map<String, dynamic> json) {
    return LoginDTO(
      userId: json['userId'],
      userPw: json['userPw'],
      platform: json['platform'] ?? "", 
      fcmToken: json['fcmToken'],
    );
  }

  // 객체 -> JSON 변환
  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'userPw': userPw,
      'platform': platform,
      'fcmToken': fcmToken,
    };
  }
}

File: buck_tanley/src/main/java/com/example/buck_tanley/domain/dto/LoginDTO.java
package com.example.buck_tanley.domain.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class LoginDTO {
    private String userId;
    private String userPw;
    private String platform;
    private String fcmToken;
}
